std/string.d

```diff
@@ -2267,7 +2239,7 @@ unittest
         assert(s2 !is s1);
         s1 = to!S("\u0131 \u0130");
         s2 = capitalize(s1);
-        assert(cmp(s2, "\u0049 \u0069") == 0);
+        assert(cmp(s2, "\u0049 i\u0307") == 0);
```

Comment:
Do you have a reference I can read to learn why this is correct (and why the old behavior was not)? This merge looks good but I'm not sure how to review this behavior change.